x/circuit/keeper/msg_server.go

```diff
@@ -0,0 +1,219 @@
+package keeper
+
+import (
+	"bytes"
+	context "context"
+	fmt "fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/circuit/types"
+)
+
+type msgServer struct {
+	Keeper
+}
+
+var _ types.MsgServer = msgServer{}
+
+// NewMsgServerImpl returns an implementation of the bank MsgServer interface
+// for the provided Keeper.
+func NewMsgServerImpl(keeper Keeper) types.MsgServer {
+	return &msgServer{Keeper: keeper}
+}
+
+func (srv msgServer) AuthorizeCircuitBreaker(goCtx context.Context, msg *types.MsgAuthorizeCircuitBreaker) (*types.MsgAuthorizeCircuitBreakerResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	address, err := srv.addressCodec.StringToBytes(msg.Granter)
+	if err != nil {
+		return nil, err
+	}
+
+	// if the granter is the module authority no need to check perms
+	if !bytes.Equal(address, srv.GetAuthority()) {
+		// Check that the authorizer has the permission level of "super admin"
+		perms, err := srv.GetPermissions(ctx, address)
+		if err != nil {
+			return nil, err
+		}
+
+		if perms.Level != types.Permissions_LEVEL_SUPER_ADMIN {
+			return nil, fmt.Errorf("only super admins can authorize users")
```

Comment:
i avoid writing the errors folder since the regestering of numbers is something we should avoid and deprecate since the error code is used in consensus and can cause unforseen issues if users swap things